packages/material-ui/src/AppBar/AppBar.js

```diff
@@ -100,17 +100,29 @@ const AppBarRoot = experimentalStyled(Paper, {
       backgroundColor: 'transparent',
       color: 'inherit',
     }),
+    ...(theme.palette.mode === 'dark' &&
+      styleProps.skipColorOnDark && {
+        backgroundColor: null,
+        color: null,
+      }),
   };
 });
 
 const AppBar = React.forwardRef(function AppBar(inProps, ref) {
   const props = useThemeProps({ props: inProps, name: 'MuiAppBar' });
-  const { className, color = 'primary', position = 'fixed', ...other } = props;
+  const {
+    className,
+    color = 'primary',
+    position = 'fixed',
+    skipColorOnDark = true,
```

Comment:
Default boolean value can only be false, per https://material-ui.com/guides/api/#property-naming. For instance, this would work:

```suggestion
    enableColorOnDark = false,
```

Reply:
Yeah, that make sense.